### PR TITLE
Add regression test for integer literals in generic arguments in where clauses

### LINKED ISSUE #67753: Fn traits with array args no longer work with const_generics enabled
```rust
#![feature(const_generics)]

fn main() {
    takes_closure_of_array_3(|[x, y, z]| {
        println!("args: x={}, y={}, z={}", x, y, z);
    });
}

fn takes_closure_of_array_3<F>(f: F)
where
    F: Fn([i32; 3]),
{
    f([1, 2, 3]);
}
```

([playpen](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=dd61053708a51e96786e54286e048ced))

Gives the error:
```
   Compiling playground v0.0.1 (/playground)
warning: the feature `const_generics` is incomplete and may cause the compiler to crash
 --> src/main.rs:1:12
  |
1 | #![feature(const_generics)]
  |            ^^^^^^^^^^^^^^
  |
  = note: `#[warn(incomplete_features)]` on by default

error[E0391]: cycle detected when const-evaluating + checking `takes_closure_of_array_3::{{constant}}#0`
  --> src/main.rs:11:17
   |
11 |     F: Fn([i32; 3]),
   |                 ^
   |
note: ...which requires const-evaluating + checking `takes_closure_of_array_3::{{constant}}#0`...
  --> src/main.rs:11:17
   |
11 |     F: Fn([i32; 3]),
   |                 ^
......................
9  | / fn takes_closure_of_array_3<F>(f: F)
10 | | where
11 | |     F: Fn([i32; 3]),
12 | | {
13 | |     f([1, 2, 3]);
14 | | }
   | |_^

```

This compiles fine with const_generics not enabled, and is a regression

cc @centril @oli-obk

### PR DESCRIPTION
**Pull request by varkor**
Closes https://github.com/rust-lang/rust/issues/67753.

### CODE REVIEW

File: src/test/ui/const-generics/integer-literal-generic-arg-in-where-clause.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// check-pass
+
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+fn takes_closure_of_array_3<F>(f: F) where F: Fn([i32; 3]) {
```
Comment (Centril) [inline src/test/ui/const-generics/integer-literal-generic-arg-in-where-clause.rs]: Can you extend this regression test to APIT and RPIT as well?

Comment (varkor) [reply src/test/ui/const-generics/integer-literal-generic-arg-in-where-clause.rs]: Something like that?

Comment (Centril) [reply src/test/ui/const-generics/integer-literal-generic-arg-in-where-clause.rs]: Yep, thanks.

**Pull request merged**